map/src/main/kotlin/edu/artic/map/MapFragment.kt

```diff
@@ -72,8 +64,12 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
     private lateinit var buildingGroundOverlay: GroundOverlay
     private var groundOverlayGenerated: Subject<Boolean> = BehaviorSubject.createDefault(false)
     private var mapClicks: Subject<Boolean> = PublishSubject.create()
-
-    private val tour: ArticTour? get() = requireActivity().intent?.extras?.getParcelable(MapActivity.ARG_TOUR)
+    private var leaveTourDialog: AlertDialog? = null
+    private var tour: ArticTour?
+        get() = requireActivity().intent?.extras?.getParcelable(MapActivity.ARG_TOUR)
+        set(value) {
+            requireActivity().intent?.extras?.putParcelable(MapActivity.ARG_TOUR, value)
```

Comment:
Please switch to `intent?.putExtra` - that properly handles the null check for the `extras` property. See https://github.com/aosp-mirror/platform_frameworks_base/blob/1c77a11db2/core/java/android/content/Intent.java#L8575

Reply:
Now uses `intent?.putExtra`.